etc/refork/build_inventory.py

```
@@ -0,0 +1,191 @@
+    v29_by_base = defaultdict(list)
+    for p in v29_tree:
+        v29_by_base[os.path.basename(p)].append(p)
+
+    loc = {}
+    for line in numstat.splitlines():
+        add, dele, rest = line.split("\t", 2)
+        if "=>" in rest:  # rename: "old => new" possibly with {…} braces
+            rest = rest.split("\t")[-1]
+        loc[rest] = (add, dele)
+
+    def fate(old_path, new_path):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `fate`

Add return type annotation: `str`

(ANN202)